.github/workflows/ci-publish.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish to NPM
+on: 
+  push:
+    tags:
+      - '*'
+
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: install
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: test
+
+  publish-npm:
+    needs: build
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: build
+      - uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: publish
+        env:
+          NPM_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```

Comment:
seems this is exactly how we pass secrets https://github.com/libero/reviewer-client/blob/master/.github/workflows/ci.yml#L98-L100